lib/house.rb

```diff
@@ -0,0 +1,50 @@
+class House
+  attr_reader :address, :rooms
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+    @market_average = 500000
+  end
+
+  def price
+    (@price.delete("$")).to_i
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    return true if price > @market_average
+      false
+  end
+
+  def rooms_from_category(category)
+    rooms.find_all do |room|
+      category == room.category
+    end
+  end
+
+  def area
+    room_areas = rooms.map { |room| room.area }
+    room_areas.sum
+  end
+
+  def details
+    details = Hash.new
+    details["price"] = price
+    details["address"] = address
+    details
+  end
+
+  def price_per_square_foot
+    (price.to_f / area.to_f).round(2)
+  end
+
+  def rooms_sorted_by_area
+    room_areas = (rooms.map { |room| room.area }).sort.reverse
```

Comment:
This doesn't work quite right, since you're mapping for the areas of the different rooms, you're going to get an array of integers instead of an array of rooms, which is not what you want.   `sort` or `sort_by` (fundamentally very similar) takes a block in which you can give instructions as to how to sort. 

```
def rooms_sorted_by_area
    rooms.sort_by {|room| room.area}.reverse
end
```

Reply:
I was running out of time and didn't get a chance to finish that up.  After I saw the result in PRY I realized I was going down the wrong path, but instead of continuing I went ahead and merged my "work" branch and made my PR at 10:59!